runtime/connection_cache.go

```diff
@@ -257,13 +257,19 @@ func (c *connectionCache) openAndMigrate(ctx context.Context, instanceID, driver
 	}
 
 	handle, err := drivers.Open(driver, config, shared, activityClient, logger)
+	if err == nil && ctx.Err() != nil {
```

Comment:
This doesn't handle timeout fully. Is that expected ?

Reply:
Yeah, it's not ideal, but `drivers.Open` doesn't currently accept a `ctx`, and anyway, DuckDB (which is the main issue for slow opens) doesn't support cancellation during opening as far as I know. So not much more we can do :(